.pipelines/stages/jobs/py-validation-job.yml

```diff
@@ -0,0 +1,217 @@
+        value: '3.9'
+    ${{ elseif eq(parameters.ep, 'cuda') }}:
+      value: '3.10'
+    ${{ elseif eq(parameters.ep, 'directml')}}:
+      value: '3.11'
+    ${{ else }}:
+      value: '3.9'
+
+  - name: pip_package_name
+    ${{ if eq(parameters.ep, 'cpu') }}:
+      value: 'onnxruntime_genai'
+    ${{ elseif eq(parameters.ep, 'cuda') }}:
+      value: 'onnxruntime_genai_cuda'
+    ${{ elseif eq(parameters.ep, 'directml')}}:
+      value: 'onnxruntime_genai_directml'
+    ${{ else }}:
+      value: 'onnxruntime_genai'
+
+  - name: prebuild_phi3_mini_model_folder
+    ${{ if eq(parameters.ep, 'cpu') }}:
+      value: 'cpu_and_mobile/cpu-int4-rtn-block-32-acc-level-4'
+    ${{ elseif eq(parameters.ep, 'cuda') }}:
+      value: 'cuda/cuda-int4-rtn-block-32'
+    ${{ elseif eq(parameters.ep, 'directml')}}:
+      value: 'directml/directml-int4-awq-block-128'
+    ${{ else }}:
+      value: 'cpu_and_mobile/cpu-int4-rtn-block-32-acc-level-4'
+
+  steps:
+  - checkout: self
+    clean: true
+    path: onnxruntime-genai
+    submodules: recursive
+
+  - script: |
+      MKDIR $(Agent.ToolsDirectory)\Python\3.12.3\arm64
+      XCOPY /s /y /h /e /c /q "C:\Python\Python312\*.*" $(Agent.ToolsDirectory)\Python\3.12.3\arm64\
+      COPY NUL $(Agent.ToolsDirectory)\Python\3.12.3\arm64.complete
+      DIR $(Agent.ToolsDirectory)\Python
+      DIR $(Agent.ToolsDirectory)\Python\3.12.3
+      DIR $(Agent.ToolsDirectory)\Python\3.12.3\arm64
+      DIR "C:\Python"
+    displayName: Copy python 3.12.3 version to agent tools directory
+    condition: and(eq(variables['arch'], 'arm64'), eq(variables['os'], 'win'), eq(variables['py_dot_ver'], '3.12'))
+
+  - script: |
+      MKDIR $(Agent.ToolsDirectory)\Python\3.11.0\arm64
+      XCOPY /s /y /h /e /c /q "C:\Python\Python311\*.*" $(Agent.ToolsDirectory)\Python\3.11.0\arm64\
+      COPY NUL $(Agent.ToolsDirectory)\Python\3.11.0\arm64.complete
+      DIR $(Agent.ToolsDirectory)\Python
+      DIR $(Agent.ToolsDirectory)\Python\3.11.0
+      DIR $(Agent.ToolsDirectory)\Python\3.11.0\arm64
+      DIR "C:\Python"
+    displayName: Copy python 3.11.0 version to agent tools directory
+    condition: and(eq(variables['arch'], 'arm64'), eq(variables['os'], 'win'), eq(variables['py_dot_ver'], '3.11'))
+
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: $(py_dot_ver)
+      addToPath: true
+      architecture: $(arch)
+
+  - ${{ if eq(parameters.os, 'linux') }}:
+    - template: steps/utils/flex-download-pipeline-artifact.yml
+      parameters:
+        StepName: 'Download Python Wheel Artifacts'
+        ArtifactName: $(ArtifactName)
+        TargetPath: '$(Build.BinariesDirectory)/wheel'
+        SpecificArtifact: ${{ parameters.specificArtifact }}
+        BuildId: ${{ parameters.BuildId }}
+  - ${{ if eq(parameters.os, 'win') }}:
+    - template: steps/utils/flex-download-pipeline-artifact.yml
+      parameters:
+        StepName: 'Download Python Wheel Artifacts'
+        ArtifactName: $(ArtifactName)-wheel
```

Comment:
The artifact name should be same between windows & linux build. But currently it's not. See:

https://github.com/microsoft/onnxruntime-genai/blob/a6bec83bf02667f6a9c6aa6340842f46e001578c/.pipelines/stages/jobs/steps/capi-win-step.yml#L161

And:

https://github.com/microsoft/onnxruntime-genai/blob/a6bec83bf02667f6a9c6aa6340842f46e001578c/.pipelines/stages/jobs/steps/capi-linux-step.yml#L159

Reply:
It depends. 
Az pipeline has two kinds of artifacts: 
1. https://learn.microsoft.com/en-us/azure/devops/pipelines/artifacts/pipeline-artifacts?view=azure-devops&tabs=yaml
2. https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/publish-build-artifacts-v1?view=azure-pipelines

If you use the second one, different jobs can use the same job. The first one doesn't. However, the second one should be deprecated.

Reply:
So the currently used approach is considered deprecated? I guess this is out of the scope of this PR, though. All I want is a consistent way to retrieve the artifacts.